### ISSUE TITLE
flutter PageRouteBuilder and CupertinoPageRoute different behavior

### ISSUE THREAD

**Issue by jinfagang** [flutter/flutter#75232]
I just want post this issue (possibabaly it's my lack of knowledge but this question has been cost my 2 DAYS).

I have 2 exactly same page which just different name for testing:

```dart
var _res = await Navigator.push(
          context,
          PageRouteBuilder(
            opaque: false,
            pageBuilder: (_, __, ___) => BlocProvider(
              bloc: blinkBloc,
              child: BlinkDetailPage3(tasks),
            ),
          ));

      // var _res = await Navigator.push(
      //     context,
      //     CupertinoPageRoute<bool>(
      //       builder: (context) => BlocProvider(
      //         bloc: blinkBloc,
      //         child: BlinkDetailPage(tasks),
      //       ),
      //     ));
```

this first one using `PageRouteBuilder` and second one using `CupertinoPageRoute`, my problem is that: **same target page without any differences but the first one, returned back and my data doesn't update, second one is normal!!!**.

This is very strange, since my bloc will called after page return, `blinkBloc.refresh();`, the when using `PageRouteBuilder` , the data doesn't refreshed.

I don't know what's the reason behind this or maybe flutter team also dont know, but I just occured it and really want to know what's happening. *what caused these 2 behavior not same and caused my data in bloc doesn update?*.

To clarify my question, the data in bloc is updates from sqflite, and if I manually refresh it, it can refresh. so I suppose the first Route way doesn't called this line when returned: 

```
var _res = await Navigator.push(
          context,
          PageRouteBuilder(
            opaque: false,
            pageBuilder: (_, __, ___) => BlocProvider(
              bloc: blinkBloc,
              child: BlinkDetailPage3(tasks),
            ),
          ));

blinkBloc.refresh();
```

If so, my question would be, why? why it designed different behavior then normal Route.
If not, does anyone build flutter can kindly give my some suggestions? this really stucked my 2 whole days!

**Comment by darshankawar** [comment]
@jinfagang 
Does the issue persist if you don't use `Bloc` ? Can you try with core `PageRouteBuilder` and `CupertinoPageRoute` without using 3rd party plugin and check if you get same result ?

**Comment by darshankawar** [comment]
@jinfagang 
As mentioned earlier as well, since you are using `bloc` which is 3rd party, it could be (not sure) be an issue at their end. That's the reason I suggested you to build a sample code using core framework implementation (without using bloc) and see if you still get same behavior.
It's important to segregate 3rd party implementation to understand where exactly the issue is.

**Comment by jinfagang** [comment]
@darrenaustin Yes you are right

I ended up deprecated PageRouterBuilder and my last conclusion is (for later people might occured this):

it's not my bloc doesn't refresh data, it's not about bloc it self, I even tried call build() function to force build widget, but the StreamBuilder doesn't update new stream, so that my _buildTaskList doesn't re-build since it needs snapshot.hasData.

I ended up give up using it, I original intention to using it is that I want set `opaque` to false, and get a transperent Route so that I can get blured effect in the next page.

Anyway, just gived it up without a solution.
